x-pack/plugins/ml/public/components/controls/select_interval/select_interval.js

```diff
@@ -38,12 +38,18 @@ function optionValueToInterval(value) {
   return interval;
 }
 
+// This service will be populated by the corresponding angularjs based one.
+export const mlSelectIntervalService = {
```

Comment:
these shared services feel a little un-react like.
would keeping the state in a root component which then propagates through props be better?
although, probably not for this PR.

Reply:
having had this all explained to me, I now have a better understanding of how this works.
so please ignore this comment